src/notebooks/controllers/notebookControllerManager.ts

```diff
@@ -441,11 +461,7 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
         let defaultPythonLanguageKernel: VSCodeNotebookController | undefined;
         controllers.forEach((item) => {
             // Sort out interactive or non-interactive controllers
-            if (
-                item.connection.kind !== 'startUsingRemoteKernelSpec' ||
-                (notebookType === 'jupyter-notebook' && item.id.includes(InteractiveControllerIdSuffix)) ||
-                (notebookType === 'interactive' && !item.id.includes(InteractiveControllerIdSuffix))
-            ) {
+            if (item.connection.kind !== 'startUsingRemoteKernelSpec') {
```

Comment:
Looks like this line is covered by line 445 above now?

Reply:
Yes, however types a funky, if i don't add this, then the item could be any controller type.
This is because item is a IVSCodeNotebookController and it has a property of type KenelConnectionMetadata, which could be local or remote.
even though we filter it out earlier, the type system doesn't know that